data_structures/src/transaction_factory.rs

```diff
@@ -236,9 +236,14 @@ pub fn calculate_weight(
 }
 
 /// Get total balance
-pub fn get_total_balance(all_utxos: &UnspentOutputsPool, pkh: PublicKeyHash) -> u64 {
+pub fn get_total_balance(
+    all_utxos: &UnspentOutputsPool,
+    old_all_utxos: Option<UnspentOutputsPool>,
+    pkh: PublicKeyHash,
+) -> (u64, u64) {
```

Comment:
Instead of `(u64, u64)` let's use a struct with named fields, similar to `WalletBalance` from the wallet

```suggestion
) -> NodeBalance {
```